src/java/org/broadinstitute/dropseqrna/utils/SplitBamByCell.java

```diff
@@ -106,18 +155,52 @@ protected int doWork() {
             writeReport(writerInfoList);
         }
 
+        if (DELETE_INPUTS) {
+            deleteInputBamFiles();
+        }
+
         return 0;
     }
 
-    private SAMFileInfo createWriterInfo(final SAMFileHeader header, int writerIdx) {
-        final String outputPath = OUTPUT.toString().replace(OUTPUT_SLUG, String.valueOf(writerIdx));
-        final File samFile = new File(outputPath);
+    private File getRelativeSplitBamFile(int splitIdx) {
+        final String outputPath = OUTPUT.toString().replace(OUTPUT_SLUG, String.valueOf(splitIdx));
+        return new File(outputPath);
+    }
+
+    private File getActualSplitBamFile(File samFile) {
         final File actualFileToOpen;
         if (OUTPUT_LIST == null) {
             actualFileToOpen = samFile;
         } else {
             actualFileToOpen = FileListParsingUtils.resolveFilePath(OUTPUT_LIST.getParentFile(), samFile);
         }
+
+        return actualFileToOpen;
+    }
+
+    private void checkOutputOverwrites() {
+        for (int splitIdx=0; splitIdx<NUM_OUTPUTS; splitIdx++) {
+            final File splitBamFile = getActualSplitBamFile(getRelativeSplitBamFile(splitIdx));
+            if (splitBamFile.exists()) {
+                // Check that this output BAM is not one of the INPUT BAMs
+                for (File inputBam : INPUT) {
+                    if (splitBamFile.getAbsolutePath().equals(inputBam.getAbsolutePath())) {
+                        throw new IllegalArgumentException("Output BAM file " + splitBamFile.getAbsolutePath() + " is the same as input BAM " + inputBam.getAbsolutePath());
+                    }
+                }
+
+                if (OVERWRITE_EXISTING) {
+                    System.out.println("BAM file " + splitBamFile.getAbsolutePath() + " exists and will be overwritten");
```

Comment:
For a warning like this, our convention is to create a static htsjdk.samtools.util.Log member of the class, and then report this using LOG.warning().  That way it goes to stderr, and user can control verbosity.  You can find examples of our using Log in the existing code.